### PR TITLE
Add a workflow to build docker images.

### PR DESCRIPTION
**Pull request by yseto**
Add a workflow to build docker images.

In order to check the operation, the execution conditions are commented.
In subsequent PRs, uncomment the commented out part. It also prepares login information for Docker Hub.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,3 +27,83 @@ jobs:
...
+      with:
+        context: .
+        file: Dockerfile.multistage
+        target: container-agent-with-plugins
+        platforms: |
+          linux/amd64
+          linux/arm64
+        # push: true
+        tags: |
+          mackerel/mackerel-container-agent:${{ steps.bump_version.outputs.VERSION }}-plugins-alpha
```
Comment (yseto) [inline .github/workflows/ci.yml:75]: Corresponds to the description here

https://github.com/mackerelio/mackerel-container-agent/blob/06d5a4b6716e88a69c28eca7ef895c41e9e90775/.circleci/config.yml#L145-L150

https://github.com/mackerelio/mackerel-container-agent/blob/06d5a4b6716e88a69c28eca7ef895c41e9e90775/.circleci/config.yml#L85-L101

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,3 +27,83 @@ jobs:
...
+#     with:
+#       context: .
+#       file: Dockerfile.multistage
+#       target: container-agent-with-plugins
+#       platforms: |
+#         linux/amd64
+#         linux/arm64
+#       push: true
+#       tags: |
+#         mackerel/mackerel-container-agent:${{ steps.tags_version.outputs.VERSION }}-plugins
```
Comment (yseto) [inline .github/workflows/ci.yml:108]: Corresponds to the description here

https://github.com/mackerelio/mackerel-container-agent/blob/06d5a4b6716e88a69c28eca7ef895c41e9e90775/.circleci/config.yml#L152-L159

https://github.com/mackerelio/mackerel-container-agent/blob/06d5a4b6716e88a69c28eca7ef895c41e9e90775/.circleci/config.yml#L119-L131

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,3 +27,83 @@ jobs:
...
+    # - uses: docker/login-action@v1
+    #   with:
+    #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+    #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Get a Bump Version Name from branch name
+      id: bump_version
+      # if: startsWith(github.ref, 'refs/heads/bump-version-')
+      # run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/heads\/bump-version-/v}
+      run: echo ::set-output name=VERSION::rehearsal
```
Comment (yseto) [inline .github/workflows/ci.yml:47]: for tests.
I'll revert the change later.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -26,3 +27,85 @@ jobs:
     - run: make lint
     - run: make test
 
+  docker:
```
Comment (susisu) [inline .github/workflows/ci.yml:30]: `docker` feels a bit ambiguous (do *what* with docker?).
How about `build` or `build-and-push`?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -26,3 +27,85 @@ jobs:
...
+    needs: [test]
+    # if: startsWith(github.ref, 'refs/heads/bump-version-') || startsWith(github.ref, 'refs/tags/v')
```
Comment (susisu) [inline .github/workflows/ci.yml:32]: Maybe the execution condition of each job/step will be clearer if `if` is placed right under the name.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -26,3 +27,92 @@ jobs:
     - run: make lint
     - run: make test
 
+  build-and-push-alpha:
+    # if: startsWith(github.ref, 'refs/heads/bump-version-')
```
Comment (susisu) [inline .github/workflows/ci.yml:31]: nice :+1:

**Review by susisu — APPROVED** [review]
LGTM!

**Comment by yseto** [comment]
Thanks for your review!

**Pull request merged**